### PR TITLE
#455 Fix bug with tricky unicode symbols

### PR DESCRIPTION
**Pull request by dyus**
add test for the bug
add regex lib which supports unicode 12.1.0 standard
replace re usage in project in favor to regex

### CODE REVIEW

**Comment by hugovk** [comment]
Also add `regex` to `install_requires`?

https://github.com/psf/black/blob/d9e71a75ccfefa3d9156a64c03313a0d4ad981e5/setup.py#L36-L42

File: setup.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def get_long_description() -> str:
         "appdirs",
         "toml>=0.9.4",
         "typed-ast>=1.3.1",
+        "regex==2019.8.19",
```
Comment (hugovk) [inline setup.py]: I think the idea is that end-user, non-dev dependencies should not be pinned and should be as loose as possible. Does it work like this?

```suggestion
        "regex",
```

Comment (dyus) [reply setup.py]: yes it should work, i'll change

**Comment by dyus** [comment]
#455

**Comment by zsol** [comment]
Thank you!

**Comment by jezdez** [comment]
The major downside of this new dependency is that it requires a C compiler during installation. Is this really the only way to fix this?

**Pull request merged**